text/0101-new-scale-control-model-based-on-IO-size.md

```diff
@@ -0,0 +1,90 @@
+
+### How to ensure fairness 
+The new scheduler policy does not guarantee that the snapshot sending size of all stores is exactly the same, but will switch to the next candidate region when the store has no available sending size. During the prototype testing process on the scale-out and scale-in scene , the sending size on all TIKV reaches the bottleneck. 
+
+### Scheduler 
+#### Balance Region Scheduler 
+The scheduler picks the store as source first, then picks the region on the source store, finally picks the target store. On the original basis, it is necessary to check the target region leader whether there is enough sending space to send. 
+#### High Priority Scheduler 
+Some scheduler may create a highest priority operator such as balance hot region, region check. For the highest operators (such as hot region operator), they can use this external space if the normal space is not enough. The default external space is about 20% of the send window. This can prevent this window size from being completely occupied by low priority task.
+
+```
+|------- Used -------|
+                     |--------Available -----------|
+|------ Normal ------------|---- Privilege —-------|
+```
+As shown in the 
+
+
+## Compatibility
+### Store Limit
+The store limiter mainly has the following functions that cannot be replaced:
+- Limit the operator generating speed. In some scenarios, users want to limit the scheduling speed to reduce the impact on workload. Currently, there is no good solution to modify the tikv configuration from PD.
+- The second highest (lowest) score store can still be scheduled. An instance cannot continuously be the source or target to be scheduler because the token is limited in the short term.
+- Remove too many peers from the same store can bring major compaction, which affect the cluster performance.
+### Multi RocksDB
+In this design, the generation of the snapshot will use check-point hard link to replace the scanning of RocksDB. It can reduce the CPU loads and the duration of snapshot generator. But the snapshots are not compressed ,the network and disk bandwidth will increase. 
+The snapshot synchronization process is basically the same, some tasks need to wait to be executed when the store has many snapshot tasks. The waiting time is still positively related to the backlog of the snapshot task, so feedback mechanism still satisfies this situation. 
+## Alternatives
+### Follower Replica 
+In this design, a new mechanism in raft algorithm will be introduced which allows the follower to replaica log or send snapshot to another peer. Since the removed peer can also be the snapshot sender, balance region scheduler no need to consult the region leader whether there is some available sending size. 
```

Comment:
```suggestion
In this design, a new mechanism in the raft algorithm will be introduced which allows the follower to replica the log or send snapshot to another peer. Since the removed peer can also be the snapshot sender, balance region scheduler no need to consult the region leader whether there is some available sending size. 
```

Reply:
fix